prefect_azure/aci.py

```diff
@@ -0,0 +1,335 @@
+            "state of an Azure Container Instances task."
+        ),
+    )
+
+    @sync_compatible
+    async def run(self, task_status: Optional[TaskStatus] = None) -> ACITaskResult:
+        """
+        Run the configured task on ACI.
+        """
+        if not self.command:
+            raise ValueError("Container cannot be run with empty command.")
+
+        # TODO: determine how to make DefaultAzureCredential work as expected
+        # if self.azure_credentials:
+        #     self.azure_credentials.login()
+        # token_credential = DefaultAzureCredential
+        token_credential = ClientSecretCredential(
+            tenant_id=self.azure_credentials.tenant_id.get_secret_value(),
+            client_id=self.azure_credentials.client_id.get_secret_value(),
+            client_secret=self.azure_credentials.client_secret.get_secret_value(),
+        )
+        aci_client = ContainerInstanceManagementClient(
+            credential=token_credential,
+            subscription_id=self.subscription_id.get_secret_value(),
+        )
+        container = self._configure_container(token_credential)
+        container_group = self._configure_container_group(token_credential, container)
+        created_container_group = None
+
+        try:
+            # Create the container group and wait for it to start
+            creation_status_poller = aci_client.container_groups.begin_create_or_update(
+                self.azure_resource_group_name, container.name, container_group
+            )
+            created_container_group = await run_sync_in_worker_thread(
+                self._wait_for_task_container_start, creation_status_poller
+            )
+
+            # If creation succeeded, group provisioning state should be 'Succeeded'
+            # and the group should have a single container
+            if (
+                created_container_group.provisioning_state == "Succeeded"
+                and len(created_container_group.containers) == 1
+            ):
+                if task_status:
+                    task_status.started(value=container.name)
+                status_code = await run_sync_in_worker_thread(
+                    self._watch_task_and_get_exit_code,
+                    aci_client,
+                    created_container_group,
+                )
+            else:
+                status_code = -1
+
+        finally:
+            if created_container_group:
+                aci_client.container_groups.begin_delete(
+                    resource_group_name=self.azure_resource_group_name,
+                    container_group_name=created_container_group.name,
+                )
+
+        return ACITaskResult(identifier=container.name, status_code=status_code)
+
+    def preview(self) -> str:
+        return ""
+
+    def _configure_container(self, credential: TokenCredential) -> Container:
+        # setup container environment variables
```

Comment:
At the moment, all methods and functions require a short docstring, or else the CI linting, particularly interrogate, will fail.

Reply:
Thanks - added it to the PR's to-do list